src/tools/web/elektrad/middleware.go

```diff
@@ -0,0 +1,137 @@
+package main
```

Comment:
Please give some overview of the architecture (in README.md). What is the middleware doing?